src/plugins/usage_collection/server/usage_collection.mock.ts

```diff
@@ -37,25 +41,37 @@ export const createUsageCollectionSetupMock = () => {
     // @ts-ignore jest.fn doesn't play nice with type guards
     isUsageCollector: jest.fn(),
     makeCollectorSetFromArray: jest.fn(),
-    makeStatsCollector: jest.fn(),
     map: jest.fn(),
     maximumWaitTimeForAllCollectorsInS: 0,
     some: jest.fn(),
     toApiFieldNames: jest.fn(),
     toObject: jest.fn(),
-    makeUsageCollector: jest.fn(),
+    makeStatsCollector: jest.fn().mockImplementation((cfg) => new Collector(logger, cfg)),
```

Comment:
An aside: We're being explicit about defining a different `UsageCollector` class from the `Collector` class and I wonder if it would be more consistent if we had a base `Collector` class and extended it to a new `StatsCollector` class even if we're not actually extending the `Collector` 🤔

Reply:
I like that. Do you mind if we do that in a separate PR? This one is already too big to my taste :(

Reply:
I've created the issue https://github.com/elastic/kibana/issues/83522 to track it